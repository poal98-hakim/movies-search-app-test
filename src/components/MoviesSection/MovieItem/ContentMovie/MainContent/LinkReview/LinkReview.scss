@use '../../../../../../scss/mixins';

.subtitle{
  display: inline-block;
  font-size: .8em;
  color: #ECE322;
  margin-top: 5px;
  margin-bottom: 10px;
  font-weight: 900;
  max-width: 400px;
  @include mixins.ellipses;
  
  @media only screen and (max-width: 750px) {
    font-size: .7em;
    max-width: 200px;
    margin: 2px 0;
  }
}