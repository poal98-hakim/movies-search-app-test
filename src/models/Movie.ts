export interface IApiMovies{
    results: IMovie[];
    has_more: boolean;
    num_results: number;
}

export interface IMovie{
    display_title: string;
    mpaa_rating: string,
    critics_pick: number;
    byline: string,
    headline: string;
    summary_short: string;
    publication_date: string,
    opening_date: string;
    date_updated: string;
    link: ILink;
    multimedia: IMultimedia;
}

export interface ILink {
    type: string;
    url: string;
    suggested_link_text: string;
}

export interface IMultimedia {
    type: string;
    src: string;
    height: number;
    width: number;
}