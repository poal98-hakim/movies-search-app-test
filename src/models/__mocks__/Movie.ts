export const MOVIE1 = {
    display_title: "Everything Everywhere All at Once",
    mpaa_rating: "R",
    critics_pick: 1,
    byline: "A.O. Scott",
    headline: "‘Everything Everywhere All at Once’ Review: It’s Messy, and Glorious",
    summary_short: "Michelle Yeoh stars as a stressed-out laundromat owner dragged into cosmic battle and genre chaos.",
    publication_date: "2022-03-24",
    opening_date: "2022-03-25",
    date_updated: "2022-03-24 16:58:03",
    link: {
        type: "article",
        url: "https://www.nytimes.com/2022/03/24/movies/everything-everywhere-all-at-once-review.html",
        suggested_link_text: "Read the New York Times Review of Everything Everywhere All at Once"
    },
    multimedia: {
        type: "mediumThreeByTwo210",
        src: "https://static01.nyt.com/images/2022/03/24/arts/24everything-everywhere/merlin_204080367_730f3977-a2fa-425e-98db-01be8fecac21-mediumThreeByTwo440.jpg",
        height: 140,
        width: 210
    }
}
export const MOVIE2 = {
    display_title: "You Are Not My Mother",
    mpaa_rating: "R2",
    critics_pick: 1,
    byline: "Jeannette Catsoulis",
    headline: "‘You Are Not My Mother’ Review: Parental Misguidance",
    summary_short: "A lonely teenager is traumatized by her mother’s volatile behavior in this impressive horror debut.",
    publication_date: "2022-03-24",
    opening_date: "2022-03-25",
    date_updated: "2022-03-24 15:32:03",
    link: {
        type: "article",
        url: "https://www.nytimes.com/2022/03/24/movies/you-are-not-my-mother-review.html",
        suggested_link_text: "Read the New York Times Review of You Are Not My Mother"
    },
    multimedia: {
        type: "mediumThreeByTwo210",
        src: "https://static01.nyt.com/images/2022/03/25/arts/mother1/mother1-mediumThreeByTwo440.jpg",
        height: 140,
        width: 210
    }
}

export const API_MOVIES_RESULT = {
    results: [MOVIE1, MOVIE2],
    num_results: 2,
    hasMore: true
}