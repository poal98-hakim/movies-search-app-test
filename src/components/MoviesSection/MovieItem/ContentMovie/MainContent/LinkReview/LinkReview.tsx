import React from "react";
import { ILink } from "../../../../../../models/Movie";
import "./LinkReview.scss";

interface IProps{
  link: ILink;
}

const LinkReview : React.VFC<IProps> = ({ link }) => {
  const {url, suggested_link_text} = link;
  return (
    <a className="subtitle" href={url} target="_blank" rel="noreferrer">{suggested_link_text}</a>
  )
}

export default LinkReview;
