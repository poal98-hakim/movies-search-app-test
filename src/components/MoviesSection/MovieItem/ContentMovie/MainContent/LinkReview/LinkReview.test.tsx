import LinkReview from "./LinkReview";
import { render, screen } from '@testing-library/react';
import { API_MOVIES_RESULT } from "../../../../../../models/__mocks__/Movie";

const PROPS = {
    link: API_MOVIES_RESULT.results[0].link
}

describe('LinkReview', () => {
  let props = {...PROPS};//using a shallow copy because I'm only reading the data. Otherwise due to the nested array I would do a deep copy

  beforeEach(() => { //before each test render the component and reset the value of props
    props = {...PROPS};
    render(<LinkReview {...props} />);
  });
  
  it('renders correctly', () => {
    const { baseElement } = render(<LinkReview {...props} />);
    expect(baseElement).toMatchSnapshot()
  })

  it('loads link review', () => {
    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", props.link.url);
    expect(link).toHaveTextContent(props.link.suggested_link_text);
  })
  
})
