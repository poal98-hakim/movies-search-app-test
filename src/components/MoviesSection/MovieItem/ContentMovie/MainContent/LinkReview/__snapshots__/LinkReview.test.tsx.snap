// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkReview renders correctly 1`] = `
<body>
  <div>
    <a
      class="subtitle"
      href="https://www.nytimes.com/2022/03/24/movies/everything-everywhere-all-at-once-review.html"
      rel="noreferrer"
      target="_blank"
    >
      Read the New York Times Review of Everything Everywhere All at Once
    </a>
  </div>
  <div>
    <a
      class="subtitle"
      href="https://www.nytimes.com/2022/03/24/movies/everything-everywhere-all-at-once-review.html"
      rel="noreferrer"
      target="_blank"
    >
      Read the New York Times Review of Everything Everywhere All at Once
    </a>
  </div>
</body>
`;
